src/source-control/SourceControl.ts

```diff
@@ -66,4 +69,10 @@ export interface SourceControl {
   attachSttackBranchesToCommits(
     commits: Commit[],
   ): Promise<Array<{ commit: Commit; sttackBranch: BranchName }>>;
+
+  /**
+   * Getter for Repo from Source Control
+   * @returns Promise that resolves into the repository
+   */
+  getRepo(): Promise<Repository>;
```

Comment:
What's this for? Won't we want to use the listener instead of a promise like this?

Reply:
Yeah, as discussed on Discord, this function is kind of redundant. Removing this.